src/openforms/submissions/status.py

```diff
@@ -23,11 +23,31 @@ class SubmissionProcessingStatus:
     submission: Submission
 
     def get_async_results(self) -> List[AsyncResult]:
+        """Retrieve the results for the task scheduled ONLY when the submission was completed."""
         if not hasattr(self, "_async_result"):
             task_ids = self.submission.on_completion_task_ids
             self._async_results = [AsyncResult(task_id) for task_id in task_ids]
         return self._async_results
 
+    def get_all_async_results(self) -> List[AsyncResult]:
+        """Retrieve the results for ALL tasks scheduled while processing a submission.
+
+        This includes tasks scheduled when the submission was completed, when it was cosigned, when a payment was
+        received or when a retry flow was triggered.
+        """
+        if not hasattr(self, "_all_async_results"):
+            task_ids_per_event = (
+                self.submission.postcompletiontasksids_set.all().values_list(
+                    "tasks_ids", flat=True
+                )
+            )
+            task_ids = []
+            for tasks_ids_list in task_ids_per_event:
+                task_ids += tasks_ids_list
```

Comment:
I wonder if you could do an aggregation query at the DB level instead? Something like ` self.submission.postcompletionmetadata_set.aggregate(models.Sum("tasks_ids"))` (untested of course)

https://docs.djangoproject.com/en/3.2/ref/contrib/postgres/aggregates/#django.contrib.postgres.aggregates.ArrayAgg might be viable

Reply:
I had tried and miserably failed. I will try again :smile: